pkg/dataflatten/string_delimited.go

```diff
@@ -0,0 +1,84 @@
+package dataflatten
+
+import (
+	"bufio"
+	"bytes"
+	"strings"
+)
+
+type dataFunc func(data []byte, opts ...FlattenOpts) ([]Row, error)
+
+type recordSplittingStrategy int
+
+const (
+	None recordSplittingStrategy = iota + 1
+	DuplicateKeys
+)
+
+func StringDelimitedFunc(kVDelimiter string, splittingStrategy recordSplittingStrategy) dataFunc {
+	switch splittingStrategy {
+	case None:
+		return singleRecordFunc(kVDelimiter)
+	case DuplicateKeys:
+		return duplicateKeyFunc(kVDelimiter)
+	default:
+		panic("Unknown record splitting strategy")
+	}
+
+}
+
+// duplicateKeyFunc returns a function that conforms to the interface expected
+// by dataflattentable.Table's execDataFunc property. properties are grouped
+// into a single record based on 'duplicate key' strategy: If a key/value pair
+// is encountered, and the record being built already has a value for that key,
+// then that record is considered 'complete'. The record is stored in the
+// collection, and a new record is started. This strategy is only suitable if
+// the data output does not exclude k/v pairs with blank/missing values, and
+// assumes that the properties for a single record are grouped together.
```

Comment:
I still think we should be more clear in this warning. 

Perhaps 
```suggestion
// collection, and a new record is started. This strategy is only suitable if
// properties for a single record are grouped together, and output includes,
// a duplicate field, indicating a new record, before any sparse values.
```